src/api/feed-discovery/test/router.test.js

```diff
@@ -0,0 +1,153 @@
+      checkValidUrl(mockReq, mockRes, mockNextFunction);
+      expect(mockNextFunction).toHaveBeenCalledTimes(1);
+      done();
+    });
+  });
+
+  describe('Test checkValidBlogPage middleware', () => {
+    it('should return 400 if the blog page responses with code other than 200', (done) => {
+      const InvalidBlogPage = 'https://notExistBlogPage.com/';
+
+      nock(InvalidBlogPage).get('/').reply(404);
+
+      request(app).post('/').send({ blogUrl: InvalidBlogPage }).expect(400, done);
+    });
+
+    it('should return 400 if the blog page responses with content type other than text/html', (done) => {
+      const InvalidBlogPage = 'https://notHtmlResponse.com/';
+
+      nock(InvalidBlogPage).get('/').reply(200, null, {
+        'Content-Type': 'text/xml',
+      });
+
+      request(app).post('/').send({ blogUrl: InvalidBlogPage }).expect(400, done);
+    });
+  });
+
+  describe('Test discoverFeedUrls middleware', () => {
+    it('should return 404 if there is no feed url discovered', (done) => {
+      const validUrl = 'https://LinkWithNoFeedUrls.com';
+      const mockBody = `
+      <html>
+        <head></head>
+        <body></body>
+      </html>
+    `;
+      const mockReq = {
+        body: {
+          blogUrl: validUrl,
+        },
+      };
+      const mockRes = {
+        locals: {
+          document: mockBody,
+        },
+      };
+      const mockNextFunction = jest.fn();
+      discoverFeedUrls(mockReq, mockRes, mockNextFunction);
+      const expectedError = createError(createError(404, 'No Feed Url Discovered'));
+      expect(mockNextFunction).toBeCalledWith(expectedError);
+      done();
+    });
+
+    it('should call next() if there is any discovered feed url', (done) => {
+      const validUrl = 'https://LinkWithFeedUrls.com';
+      const mockBody = `
+      <html>
+        <head>
+          <link rel="alternate" type="application/rss+xml" href="https://validBlog.com/feed"/>
+        </head>
+        <body></body>
+      </html>
+    `;
+      const mockReq = {
+        body: {
+          blogUrl: validUrl,
+        },
+      };
+      const mockRes = {
+        locals: {
+          document: mockBody,
+        },
+      };
+      const mockNextFunction = jest.fn();
+      discoverFeedUrls(mockReq, mockRes, mockNextFunction);
+      expect(mockNextFunction).toHaveBeenCalledTimes(1);
+      expect(mockRes.locals.feedUrls).toEqual(['https://validBlog.com/feed']);
+      done();
+    });
+  });
+
+  it('should return 200 status and if all middlewares got passed', async (done) => {
+    const blogUrl = 'https://test321.blogspot.com/';
+    const mockBlogUrlResponseBody = `
+      <!doctype html>
+      <html lang="en">
+        <head>
+          <link rel="alternate" type="application/atom+xml" href="https://test321.blogspot.com/feeds/posts/default/-/open-source"/>
+          <link rel="alternate" type="application/rss+xml" href="https://test321.blogspot.com/feeds/posts/default/-/open-source?alt=rss"/>
```

Comment:
This is a good start, but add tests for every feed type that you search for in that Array.  Make sure they all work.